akka-actor-typed-tests/src/test/java/jdocs/akka/typed/extensions/ExtensionDocTest.java

```diff
@@ -0,0 +1,72 @@
+package jdocs.akka.typed.extensions;
+
+import akka.actor.typed.ActorSystem;
+import akka.actor.typed.Behavior;
+import akka.actor.typed.Extension;
+import akka.actor.typed.ExtensionId;
+import akka.actor.typed.javadsl.Behaviors;
+
+public class ExtensionDocTest {
+
+   //#shared-resource
+   public static class ExpensiveDatabaseConnection {
+      public Object executeQuery(String query) {
+         throw new RuntimeException("I should do a database query");
+      }
+      //...
+   }
+   //#shared-resource
+
+   //#extension
+   public static class DatabaseConnectionPool implements Extension {
+      private final ExpensiveDatabaseConnection _connection;
```

Comment:
Not super realistic that it'd be final, so maybe mention that in a comment?

Reply:
How come if it is created when the extension is initialised?